### PR TITLE
fix moban.exceptions.NoThirdPartyEngine: jinja2

### PR DESCRIPTION
**Pull request by PrajwalM2212**
Closes https://github.com/moremoban/moban/issues/121

Before raising the PR, here is a check list:

1) have you written unit tests for your code changes?
2) have you run "make format"?
3) are you requesting to "dev"?
4) have you updated the change log?
5) do you think that you can understand your changes after 6 month?
   5.1) can someone else understand your changes without your explanation?
6) are you pround of your code changes?
   6.1) do you have the feeling of achievement?

### CODE REVIEW

**Comment by PrajwalM2212** [comment]
Travis is failing due to 
1.  Formatting errors.
2.  `unicode` was originally used for `handlebars` which is incompatible with python3 

I will fix those when I am free.

File: moban/engine_factory.py [hunk 1]
```diff
@@ -146,8 +116,8 @@ def verify_the_existence_of_directories(dirs):
         if os.path.exists(directory):
             continue
         should_I_ignore = (
-            constants.DEFAULT_CONFIGURATION_DIRNAME in directory
-            or constants.DEFAULT_TEMPLATE_DIRNAME in directory
+                constants.DEFAULT_CONFIGURATION_DIRNAME in directory
```
Comment (jayvdb) [inline moban/engine_factory.py]: unnecessary change

File: moban/engine_handlebars.py [hunk 2]
```diff
@@ -77,7 +93,7 @@ def _apply_template(self, template, data, output):
         template_file = self.find_template_file(template)
         with open(template_file, "r") as source:
             if sys.version_info[0] < 3:
-                template = Compiler().compile(unicode(source.read())) # noqa
+                template = Compiler().compile(unicode(source.read()))  # noqa
```
Comment (jayvdb) [inline moban/engine_handlebars.py]: unrelated change - create an issue for these pep issues, and schedule it for a beginner

File: moban/engine.py [hunk 3]
```diff
@@ -14,17 +16,47 @@
...
 from moban.engine_factory import (
-    MOBAN_ALL,
-    BUILTIN_EXENSIONS,
     Context,
     expand_template_directory,
     expand_template_directories,
-    verify_the_existence_of_directories
-)
+    verify_the_existence_of_directories,
+    Strategy)
```
Comment (jayvdb) [inline moban/engine.py]: put `)` on next line, dedented

File: moban/engine.py [hunk 4]
```diff
@@ -14,17 +16,47 @@
...
+
+from lml.plugin import PluginManager
```
Comment (jayvdb) [inline moban/engine.py]: this should go with the other lml imports

File: moban/engine.py [hunk 5]
```diff
@@ -14,17 +16,47 @@
...
+
+    def raise_exception(self, key):
+        raise exceptions.NoThirdPartyEngine(key)
 
-FILTERS = JinjaFilterManager()
-TESTS = JinjaTestManager()
-GLOBALS = JinjaGlobalsManager()
+
+ENGINES = EngineFactory()
+MOBAN_EXTENSIONS = "^moban_.+$"
```
Comment (jayvdb) [inline moban/engine.py]: belongs in `constants` ?

File: moban/engine_handlebars.py [hunk 6]
```diff
@@ -9,13 +9,12 @@
...
 from moban.engine_factory import (
-    MOBAN_ALL,
-    BUILTIN_EXENSIONS,
     Context,
     expand_template_directory,
     expand_template_directories,
-    verify_the_existence_of_directories
-)
+    verify_the_existence_of_directories,
+    Strategy)
```
Comment (jayvdb) [inline moban/engine_handlebars.py]: trailing `,` and put `)` on next line

File: moban/engine_handlebars.py [hunk 7]
```diff
@@ -39,11 +38,28 @@ def find_template_file(self, template_file):
...
+            if os.path.exists(true_template_file):
+                break
+        utils.file_permissions_copy(true_template_file, output_file)
+
     def render_to_file(self, template_file, data_file, output_file):
         template_file = self.find_template_file(template_file)
         with open(template_file, "r") as source:
             if sys.version_info[0] < 3:
-                template = Compiler().compile(unicode(source.read())) # noqa
+                template = Compiler().compile(str(source.read()))  # noqa
```
Comment (jayvdb) [inline moban/engine_handlebars.py]: this worries me.

it should be a separate commit.

File: moban/engine_factory.py (hunk 1 above)
Comment (PrajwalM2212) [reply moban/engine_factory.py]: Automatically done by formatting

**Comment by PrajwalM2212** [comment]
screen shot 2018-11-14 at 6 32 44 am
Working as usual
